libraries/AP_Telemetry/MQTT_Mavlink.cpp

```diff
@@ -0,0 +1,275 @@
+            token = strtok(&cmd[6], ",");
+            if (token != nullptr) {
+                mission_item.x = atof(token); // 16 float
+            } else {
+                return 0;
+            }
+            token = strtok(nullptr, ",");
+            if (token != nullptr) {
+                mission_item.y = atof(token); // 16 float
+            } else {
+                return 0;
+            }
+            token = strtok(nullptr, ",");
+            if (token != nullptr) {
+                mission_item.z = atof(token); // 16 float
+            } else {
+                return 0;
+            }
+        }
+        _mav_put_float(buf, 16,mission_item.x);
+        _mav_put_float(buf, 20,mission_item.y);
+        _mav_put_float(buf, 24,mission_item.z);
+
+        mission_item.seq = 0;// 28 uint16
+        _mav_put_uint16_t(buf, 28, mission_item.seq);
+        mission_item.command = 16; // 30 uint16
+        _mav_put_uint16_t(buf, 30, mission_item.command);
+        mission_item.target_system = 1; //32 uint8
+        _mav_put_uint16_t(buf, 32, mission_item.target_system);
+        mission_item.target_component = 0; // 33 uint8
+        _mav_put_uint16_t(buf, 33, mission_item.target_component);
+        mission_item.frame = 3; // 34 uint8
+        _mav_put_uint16_t(buf, 34, mission_item.frame);
+        mission_item.current = 2; // 35 uint8
+        _mav_put_uint16_t(buf, 35, mission_item.current);
+        mission_item.autocontinue = 0; // 36 uint8
+        _mav_put_uint16_t(buf, 36, mission_item.autocontinue);
+
+        memcpy(_MAV_PAYLOAD_NON_CONST(msg), buf, 40);// 30 is about
+        memset(msg, 0, sizeof(mavlink_message_t));
+        sprintf(topic, "$ardupilot/copter/quad/ack/%04d", mavlink_system.sysid);
+        tele_mqtt->send_message("Fly to received", topic);
+        ret = 1;
+    } else if (strncmp(cmd, "param set circle_radius", 23) == 0) {
+        mavlink_param_set_t packet1;
+        unsigned char system_id;
+        unsigned char component_id;
+
+        system_id = 0;
+        component_id = 0;
+        memset(&packet1, 0, sizeof(packet1));
+        packet1.param_value = atof(&cmd[23]);
+        packet1.target_system = 0;
+        packet1.target_component = 0;
+        packet1.param_type = MAV_PARAM_TYPE_REAL32;
+        strcpy(packet1.param_id, "CIRCLE_RADIUS");
+        mavlink_msg_param_set_pack(system_id,
+                                   component_id, msg ,
+                                   packet1.target_system ,
+                                   packet1.target_component ,
+                                   packet1.param_id ,
+                                   packet1.param_value ,
+                                   packet1.param_type);
+        memset(msg, 0, sizeof(mavlink_message_t));
+        sprintf(topic, "$ardupilot/copter/quad/ack/%04d", mavlink_system.sysid);
+        tele_mqtt->send_message("Param for circle received", topic);
+        ret = 1;
+    } else if (strcmp(cmd, "mqtt log_off") == 0) {
```

Comment:
The number of spaces is also limited. I think that it can be limited to the character string of the parameter by using split so that multiple spaces may be entered.